src/remind_me/cog.py

```diff
@@ -0,0 +1,123 @@
+from logging import getLogger
+import asyncio
+import datetime
+import dateparser
+import tabulate
+from asgiref.sync import sync_to_async
+from discord.ext import commands
+from discord.ext.commands import Bot, Context
+from discord_bot.cog import BaseCog
+from .models import Reminder
+
+
+logger = getLogger(__name__)
+
+
+LIST_REMINDERS_HEADERS = ["ID", "Reminder Time", "Reminder Text"]
+
+
+class RemindMeCog(BaseCog):
+    @commands.Cog.listener()
+    async def on_ready(self) -> None:
+        existing_reminders = Reminder.objects.all()
+        start_time = datetime.datetime.now(datetime.UTC)
+        # Check for any reminders that may have expired while the bot was down and send them
+        # or create timers for the remaining ones
+        async for existing_reminder in existing_reminders:
+            if existing_reminder.reminder_time <= start_time:
+                await self._send_reminder(existing_reminder)
+            else:
+                delta = (existing_reminder.reminder_time - start_time).seconds
+                self._create_timer(delta, existing_reminder)
+
+    def _create_timer(self, delta: float, reminder: Reminder) -> None:
+        logger.info(
+            "Reminding Discord User (ID=%d) in %0.2f seconds in channel %d",
+            reminder.creator_id,
+            delta,
+            reminder.initial_channel_id,
+        )
+        asyncio.get_running_loop().call_later(
+            delta, lambda: asyncio.create_task(self._send_reminder(reminder))
+        )
+
+    async def _send_reminder(self, reminder: Reminder) -> None:
+        channel = self.bot.get_partial_messageable(reminder.initial_channel_id)
+        if not channel:
+            logger.warn(
+                "No channel was found for %d, doing nothing.",
+                reminder.initial_channel_id,
+            )
+            await reminder.adelete()
```

Comment:
I assume you're missing a return here? Either way, the control flow is probably clearer if you just delete in one spot and put the send in the if block.